fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ReviewQualityCheckBottomSheet.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.compose.animation.animateContentSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckAction
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckState.OptedIn.ProductReviewState.ProductAnalysis
+import org.mozilla.fenix.shopping.state.ReviewQualityCheckStore
+
+/**
+ * Top-level UI for the Review Quality Check feature.
+ *
+ * @param store [ReviewQualityCheckStore] that holds the state.
+ * @param onRequestDismiss Invoked when a user action requests dismissal of the bottom sheet.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+fun ReviewQualityCheckBottomSheet(
+    store: ReviewQualityCheckStore,
+    onRequestDismiss: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val reviewQualityCheckState by store.observeAsState(ReviewQualityCheckState.Initial) { it }
+
+    ReviewQualityCheckScaffold(
+        onRequestDismiss = onRequestDismiss,
+        modifier = modifier.animateContentSize(),
+    ) {
+        when (val state = reviewQualityCheckState) {
```

Comment:
it may not be necessary to create a local `val` via the `when` 
```suggestion
        when (reviewQualityCheckState) {
```

Reply:
This makes smart cast possible